linux_os/guide/system/auditing/auditd_configure_rules/audit_kernel_module_loading/audit_rules_kernel_module_loading/ansible/shared.yml

```diff
@@ -35,47 +94,78 @@
   when: find_modules.matched is defined and find_modules.matched > 0
 
 - name: Inserts/replaces the modules rule in rules.d when on x86
-  lineinfile:
-    path: "{{ all_files[0] }}"
-    {{% if product == "rhel6" %}}
-    line: "-a always,exit -F arch=b32 -S init_module -S delete_module -k modules"
-    {{% else %}}
-    line: "-a always,exit -F arch=b32 -S init_module -S delete_module -S finit_module -k modules"
-    {{% endif %}}
-    create: yes
+  block:
+    - name: start the line
+      set_fact: tmpline="-a always,exit -F arch=b32 "
+    - name: add syscalls
+      set_fact: tmpline="{{tmpline + '-S ' + item.item + ' ' }}"
+      loop: "{{ audit_kernel_found_32_rules_d.results }}"
+      when: item.matched is defined and item.matched == 0
+    - name: finish the line
+      set_fact: tmpline="{{ tmpline + '-k modules' }}"
+    - name: insert/replace the line in appropriate file
+      lineinfile:
+        path: "{{ all_files[0] }}"
+        line: "{{ tmpline }}"
+        create: true
+        state: present
+  when: audit_kernel_matched_32_rules_d < audit_kernel_number_of_syscalls
 
 - name: Inserts/replaces the modules rule in rules.d when on x86_64
-  lineinfile:
-    path: "{{ all_files[0] }}"
-    {{% if product == "rhel6" %}}
-    line: "-a always,exit -F arch=b64 -S init_module -S delete_module -k modules"
-    {{% else %}}
-    line: "-a always,exit -F arch=b64 -S init_module -S delete_module -S finit_module -k modules"
-    {{% endif %}}
-    create: yes
-  when: audit_arch is defined and audit_arch == 'b64'
+  block:
+    - name: start the line
+      set_fact: tmpline="-a always,exit -F arch=b64 "
+    - name: add syscalls
+      set_fact: tmpline="{{tmpline + '-S ' + item.item + ' ' }}"
+      loop: "{{ audit_kernel_found_64_rules_d.results }}"
+      when: item.matched is defined and item.matched == 0
+    - name: finish the line
+      set_fact: tmpline="{{ tmpline + '-k modules' }}"
+    - name: insert/replace the line in appropriate file
```

Comment:
Watch out for the capitalization of the task name.
The task only adds new audit rules, it doesn't replace them.

The name of task here can be a bit more descriptive like in the block name.